core/src/main/java/org/mule/runtime/core/internal/streaming/DefaultStreamingManager.java

```diff
@@ -67,19 +68,9 @@ public ByteStreamingManager forBytes() {
     return byteStreamingManager;
   }
 
-  /**
-   * {@inheritDoc}
-   */
   @Override
-  public void success(Event event) {
-    byteStreamingManager.success(event);
+  public void registerEventContext(EventContext eventContext) {
```

Comment:
this method should not be called in all the places you're currently doing it. You should only do it in `org.mule.runtime.core.internal.streaming.bytes.factory.AbstractCursorStreamProviderFactory#of`.

Also this method should actually be a part of `org.mule.runtime.core.internal.streaming.bytes.DefaultByteStreamingManager#onOpen(org.mule.runtime.core.internal.streaming.bytes.CursorStreamProviderAdapter)`